### DISCUSSION TITLE
How to query nested categories?

### DISCUSSION THREAD

**Discussion by ShahriarKh** [wp-graphql/wp-graphql#2875] [General]
Imagine I have categories `A`, `B` and `B1` (child of `B`).

When I query categories, I got all of them at the same level:
```graphql
 categories {
    nodes {
      name
    }
  }
```
```json
"categories": {
  "nodes": [
    {
      "name": "A"
    },
    {
      "name": "B"
    },
    {
      "name": "B1" # B1 shouldn't be top-level like A and B, because it's a child category
    },
  ]
}
```
And when I query categories with children, I get duplicate categories:
```graphql
 categories {
    nodes {
      name
        children {
          nodes {
            name
          }
        }
    }
  }
```
```json
"categories": {
  "nodes": [
    {
      "name": "A",
      "children": {
        "nodes": []
      }
    },
    {
      "name": "B",
      "children": {
        "nodes": [
          {
            "name": "B1"
          },
        ]
      }
    },
    {
      "name": "B1", # B1 shouldn't be top-level like A and B, because it's a child category
      "children": {
        "nodes": []
      }
    },
  ]
}
```
How can I get categories and their sub-categories in a nested manner? I noticed that top-level categories have `parent: null`, but I can't find a way to filter categories this way... (maybe `categories(where: {parent: null}) { ... }`

**Comment by chrisgrabinski** [comment]
I believe [the "official" way](https://master--wpgraphql-docs.netlify.app/getting-started/menus/#converting-a-flat-list-to-hierarchical-list) to deal with this is converting the flat list from the query result into a hierarchical list.
It's most definitely not what you're looking for but it might solve your problem.

**Comment by ShahriarKh** [comment]
I just figured out a hacky way! By adding a specific text to category description, you can use `descriptionLike`:
```graphql
query CATEGORIES {
  categories(where: {descriptionLike: "parentForGraphql"}) {
    nodes {
      name
    }
  }
}
```
